drivers/usb/udc/udc_kinetis.c

```diff
@@ -170,25 +170,20 @@ static ALWAYS_INLINE void usbfsotg_resume_tx(const struct device *dev)
 	base->CTL &= ~USB_CTL_TXSUSPENDTOKENBUSY_MASK;
 }
 
-/* Initiate a new transfer, must not be used for control endpoint OUT */
-static int usbfsotg_xfer_start(const struct device *dev,
-			       struct udc_ep_config *const cfg)
+static int usbfsotg_xfer_continue(const struct device *dev,
+				  struct udc_ep_config *const cfg,
+				  struct net_buf *const buf)
 {
 	const struct usbfsotg_config *config = dev->config;
 	USB_Type *base = config->base;
 	struct usbfsotg_bd *bd;
-	struct net_buf *buf;
 	uint8_t *data_ptr;
 	size_t len;
 
-	buf = udc_buf_peek(dev, cfg->addr);
-	if (buf == NULL) {
-		return -ENODATA;
-	}
-
 	bd = usbfsotg_get_ebd(dev, cfg, false);
-	if (usbfsotg_bd_is_busy(bd)) {
-		LOG_DBG("ep 0x%02x buf busy", cfg->addr);
+	if (unlikely(usbfsotg_bd_is_busy(bd))) {
+		LOG_ERR("ep 0x%02x buf busy", cfg->addr);
+		__ASSERT_NO_MSG(false);
```

Comment:
Why assert here?

Reply:
Then I would prefer the assert to be moved before the if, e.g. `__ASSERT_NO_MSG(!usbfsotg_bd_is_busy(bd));`